### PR TITLE
Techdebt/forms 530 improve logging

### PR DESCRIPTION
**Pull request by MCrawleyHomeOffice**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Please also include any acceptance criteria if you have any.

- Added poni to main workspace package.json
- Replaced all console.log and console.error in designer, model and runner

## Type of change

TechDebt

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- [x] Manual smoke testing
- [x] Run automated smoke test
- [x] Run unit tests 

# Checklist:

- [x] My changes do not introduce any new linting errors
- [ x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [N/A] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [N/A] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto main and there are no code conflicts
- [N/A] I have checked deployments are working in all environments
- [N/A] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Comment by jenbutongit** [comment]
hey @MCrawleyHomeOffice - is this ready for review? happy to pick this up once you've resolved merge conflicts. Don't forget to request reviewers once you've taken them off do not review/merge

**Review by gstevenson — APPROVED** [review]

**Comment by jenbutongit** [comment]
Don't merge yet please - I want to take a look first!

File: runner/src/server/plugins/logger.ts
Comment (jenbutongit) [inline runner/src/server/plugins/logger.ts:5]: please don't use this! we already register pino to the server. This configuration is also incorrect "designer"

see `runner/src/server/plugins/logging.ts`

File: runner/src/index.ts
Comment (jenbutongit) [inline runner/src/index.ts:8]: you can just use console.error here, as suggested by hapi-pino, https://github.com/pinojs/hapi-pino#usage

File: runner/src/server/plugins/engine/services/addressService.ts
Comment (jenbutongit) [inline runner/src/server/plugins/engine/services/addressService.ts:17]: you're using pino logger, but not the one that's been registered to the server. Take a look at the constructor for `runner/src/server/services/webhookService.ts` if you want it to work in a service

File: designer/server/index.ts
Comment (jenbutongit) [inline designer/server/index.ts:8]: you can just use console.error here, as suggested by hapi-pino, https://github.com/pinojs/hapi-pino#usage

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: runner/src/server/plugins/logger.ts
Comment (MCrawleyHomeOffice) [reply runner/src/server/plugins/logger.ts:5]: Thanks, well spotted on the "designer/runner" mistake. 

To clarify the reason for this, there are some scenarios where we *don't* have a request or server available so we cannot use hapi-pino.

eg. runner/src/server/plugins/engine/services/addressService.ts

Would you rather we just did a console.log/error here instead?

File: runner/src/server/plugins/engine/services/addressService.ts
Comment (MCrawleyHomeOffice) [reply runner/src/server/plugins/engine/services/addressService.ts:17]: Thanks, I understand now.

Before I do this, can I double check the actual approach you have in mind?

We could either pass the server.logger in to the function itself as an extra parameter, or we could convert the addressService to a class and pass the logger in the constructor?

Either way, we don't appear to actually use this addressService at all right now so we won't actually be passing it in from anywhere, but worth us getting it right now for when it is in-use.

Comment (jenbutongit) [reply runner/src/server/plugins/engine/services/addressService.ts:17]: I think convert it to a class, so we can register and access the service from anywhere within the server. When we register services to the server, using schmervice, schmervice will initialise the object with `(server, name)`. 

so your constructor will look like this
```.ts
class MyService {
  constructor(server, name) { // or just (server) 
    this.logger = server.logger 
  }
}
```

or you can just delete it, since neither depts are using it :p

File: designer/client/plugins/logger.ts
Comment (MCrawleyHomeOffice) [reply designer/client/plugins/logger.ts:5]: Fair points. The AC on the ticket is simply to replace all console.logs with pino which is done, however after speaking to @gstevenson  he has confirmed we should be making sure Kibana handles these correctly too.

I'll look into your suggestions and be in touch if I need more clarification.

File: runner/src/server/plugins/engine/services/addressService.ts
Comment (MCrawleyHomeOffice) [reply runner/src/server/plugins/engine/services/addressService.ts:17]: Thanks, I think I'll try to amend it rather than remove it.

How do you feel about moving the service into the src/server/services folder so it's with the rest of the services (eg. WebhookService). It'll look cleaner when we import and register the service, but I'm unsure if there's a reason it currently lives in the engine.

File: designer/client/section-create.js
Comment (MCrawleyHomeOffice) [reply designer/client/section-create.js:27]: Agreed and done.

File: runner/src/server/plugins/engine/services/addressService.ts
Comment (jenbutongit) [reply runner/src/server/plugins/engine/services/addressService.ts:17]: Yep, that sounds good! I didn't realise it wasn't already..! 😱

File: designer/client/plugins/logger.ts
Comment (jenbutongit) [reply designer/client/plugins/logger.ts:5]: fwiw, I reckon all of the designer/runner *server* logs should be sorted with this PR (which was causing the splitting in kibana - so the reason for the ticket should be resolved now.. just thinking about your ticket scope!)

Comment (MCrawleyHomeOffice) [reply designer/client/plugins/logger.ts:5]: I've now added the transmit option which seems to be working, but I do have a number of outstanding queries I'd like to run by you (see comments in the designers logger.ts).

I will discuss via Gitter and arrange for time to address these remaining points.

File: designer/client/plugins/logger.ts
Comment (jenbutongit) [inline designer/client/plugins/logger.ts:11]: > What levels do we send to the new endpoint?
We should make it configurable by env environment - for dev environments you usually want more logging (so a lower level.. or if a dev finds it annoying would change it to something higher). For higher environments, you'd want less logging, because they're actually getting stored somewhere and might start running up your aws bill! 

> Does the endpoint url matter? eg. /transmit
not really. you can have a look for common/standards but I don't think there are any

> Should this be async / do we care about the response or do we just fire and forget?
I think the server should still send a response, with status code 204 (empty response) / 5xx if it failed(?), but the designer app itself should just fire and forget. Unless you want to retry sending them, but there's not much else the designer can do 

> Do we still need browser: {asObject: true}? If so, why?
I'd recommend it. When you do `pino.error(['MyClass', 'methodThatCausedIt'], error)`, the log will be correctly parsed. Otherwise you might have to do some additional parsing on the /transmit endpoint? could be wrong!

Comment (MCrawleyHomeOffice) [reply designer/client/plugins/logger.ts]: As discussed, we've now added an environment variable called "REACT_LOG_LEVEL" which will dictate the log level, else fall back on the NODE_ENV to determine the log level. This should allow developers to specify their own log level for the react client.

File: designer/client/plugins/logger.ts
Comment (MCrawleyHomeOffice) [reply designer/client/plugins/logger.ts]: Discussed and resolved.

**Review by jenbutongit — CHANGES_REQUESTED** [review]
LGTM! Thanks! 

one small thing, worth adding a row to https://github.com/XGovFormBuilder/digital-form-builder/edit/main/designer/README.md for your new REACT_XX env variables.

Also, could you remove pino from the root package.json? I think just the designer is using it

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**
